C7GameDataTests/SaveTest.cs

```diff
@@ -0,0 +1,128 @@
+
+	private static string getBasePath(string file) => Path.Combine(testDirectory, file);
+
+	private static string getDataPath(string file) => Path.Combine(testDirectory, "data", file);
+
+	private static string defaultBicPath {
+		get => Path.Combine(Civ3Location.GetCiv3Path(), "Conquests", "conquests.biq");
+	}
+
+	private static string testDirectory {
+		get {
+			string[] parts = AppDomain.CurrentDomain.BaseDirectory.Split(Path.DirectorySeparatorChar);
+			int pos = parts.Reverse().ToList().FindIndex(s => s == C7GameDataTestsFolderName);
+			string up = string.Concat("..", Path.DirectorySeparatorChar);
+			string relativePath = string.Concat(Enumerable.Repeat(up, pos - 1));
+			return Path.GetFullPath(relativePath);
+		}
+	}
+
+	[Fact]
+	public void SimpleSave()
+	{
+		// simple load SaveGame and save to file:
+		string outputNeverGameDataPath = getDataPath("output/static-save-never-game-data.json");
+
+		// load SaveGame but convert to and from GameData before saving to file:
+		string outputWasGameDataPath = getDataPath("output/static-save-was-game-data.json");
+
+		string developerSave = getBasePath("../C7/Text/c7-static-map-save.json");
+
+		SaveGame saveNeverGameData = SaveGame.Load(developerSave);
+
+		saveNeverGameData.Save(outputNeverGameDataPath);
+		GameData gameData = saveNeverGameData.ToGameData();
+		SaveGame saveWasGameData = SaveGame.FromGameData(gameData);
+		saveWasGameData.Save(outputWasGameDataPath);
+
+		byte[] original = File.ReadAllBytes(developerSave);
+		byte[] savedNeverGameData = File.ReadAllBytes(outputNeverGameDataPath);
+		byte[] savedWasGameData = File.ReadAllBytes(outputWasGameDataPath);
+
+		// saved files should not be empty
+		Assert.NotEmpty(savedNeverGameData);
+		Assert.NotEmpty(savedWasGameData);
+
+		// saved files should be the same as the original
+		Assert.Equal(original, savedNeverGameData);
+
+		// TODO: Currently the order of the properties in the json is different,
+		// so this fails. For testing, it would be convenient to sort SaveGame
+		// fields alphabetically before serializing to json.
+
+		Assert.Equal(original, savedWasGameData);
+	}
+
+	[Fact]
+	public void LoadGOTMWinners() {
+		string path = getDataPath("gotm");
+		try {
+			Directory.CreateDirectory(getDataPath("output"));
+		} catch {}
+
+		DirectoryInfo directoryInfo = new DirectoryInfo(path);
+		IEnumerable<FileInfo> saveFiles = directoryInfo.EnumerateFiles();
+		int i = 0;
+		foreach (FileInfo saveFileInfo in saveFiles) {
+			SaveGame game = null;
+			GameData gd = null;
+			Exception ex = Record.Exception(() => {
+				game = ImportCiv3.ImportSav(saveFileInfo.FullName, defaultBicPath);
+			});
+			Assert.Null(ex);
+			ex = Record.Exception(() => {
+				gd = game.ToGameData();
+			});
+			Assert.Null(ex);
+			Assert.NotNull(game);
+			Assert.NotNull(gd);
+			game.Save(Path.Combine(testDirectory, "data", "output", $"gotm_save_{i}.json"));
+			i++;
+		}
+	}
+
+	[Fact]
+	public void LoadAllConquests() {
```

Comment:
does this pass? I checked out the branch and tried loading the first Conquest and it crashes

Reply:
I'll look into this and see if it's passing locally